### PR TITLE
Mgmt-8280 subsystem test

### PR DESCRIPTION
**Pull request by isaacdorfman**
# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by sagidayan** [comment]
@isaacdorfman Thanks for the PR 🎉 

PR title should be prefixed with `MGMT-xyz: ` (all caps)
also, can you add a description?

File: subsystem/kubeapi_test.go [hunk 1]
```diff
@@ -855,6 +870,84 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
+		infraEnv := getInfraEnvFromDBByKubeKey(ctx, db, infraEnvKey, waitForReconcileTimeout)
+		configureLocalAgentClient(infraEnv.ID.String())
+		hosts := make([]*models.Host, 0)
+		ips := hostutil.GenerateIPv4Addresses(3, defaultCIDRv4)
+		for i := 0; i < 3; i++ {
+			hostname := fmt.Sprintf("h%d", i)
+			host := registerNode(ctx, *infraEnv.ID, hostname, ips[i])
+			hosts = append(hosts, host)
+		}
+		for _, host := range hosts {
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:904]: No need for all the stuff done from here.
I think that what we want to test is:

- Create ACI with Ignition URL and cert
- Load cluster from DB and make sure that the URL and Cert were persisted
- Add Host
- Update Host spec with ignition token
- Load host from DB and make sure token was persisted.

File: subsystem/kubeapi_test.go [hunk 2]
```diff
@@ -855,6 +870,84 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}, "2m", "2s").Should(Equal(0))
 	})
 
+	FIt("deploy CD with ACI and agents with ignitionEndpoint - wait for ready, delete CD and verify ACI and agents deletion", func() {
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+		caCertificateSecretName := "ca-certificate"
+		deploySecret(ctx, kubeClient, caCertificateSecretName, map[string]string{corev1.TLSCertKey: "abc"})
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:877]: Add a defer to delete this secret

**Comment by rollandf** [comment]
Please do not ignore the PR template

File: subsystem/kubeapi_test.go [hunk 3]
```diff
@@ -855,6 +870,84 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
+		modifiedAciSpec := aciSpec.DeepCopy()
+		modifiedAciSpec.IgnitionEndpoint = &hiveext.IgnitionEndpoint{
+			Url: "example.com",
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:880]: `            "message": "The Spec could not be synced due to an input error: The URL scheme must be http(s) and specified in the URL: 'example.com'"`

Comment (isaacdorfman) [reply subsystem/kubeapi_test.go:880]: Where did you see this error?

Comment (rollandf) [reply subsystem/kubeapi_test.go:880]: In the CI result, click `open stdout`

blob:https://prow.ci.openshift.org/6a9468b0-7d81-456f-90af-d2b0141f493c
Screenshot from 2021-12-23 12-09-00

**Comment by rollandf** [comment]
@isaacdorfman  you will need to rebase this on top of #3099 once merged for the agent part

**Comment by rollandf** [comment]
Implemented in #3093 3143

**Pull request closed without merging**